### PR TITLE
Added notify action to Fastlane plugin

### PR DESCRIPTION
**Pull request by Cawllec**
This adds a notify action for both Android and iOS to the Fastlane plugin, allowing automatic notification of a new build.
- Retrieves API KEY, APP VERSION, and RELEASE STAGE from AppManifest.xml automatically in android
- Retrieves API KEY from info.plist automatically in iOS
- Allows configuration of all parameters through Fastlane action API

### CODE REVIEW

File: tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/bugsnag_notify_build.rb [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+      end
+
+      def self.is_supported?(platform)
+        [:ios, :mac, :android].include?(platform)
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :config_file,
+                                       description: "AppManifest.xml/Info.plist location",
```
Comment (fractalwrench) [inline tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/bugsnag_notify_build.rb]: AppManifest -> AndroidManifest

**Comment by Cawllec** [comment]
## Discussion:
- Git path uses `Dir.pwd` to get git options, should this be optional
- Can more configuration options be acquired from `info.plist`/`AppManifest.xml`?

File: tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/bugsnag_notify_build.rb [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+      end
+
+      private
+
+      def self.get_android_options(params)
+        require "xmlsimple"
+        if params[:config_file].nil?
+          config_file = Dir::glob("./src/main/AndroidManifest.xml").first
+          if config_file.nil?
+            puts "Not config_file option was provided, and your AndroidManifest could not be found. Bugsnag API key could not be automatically detected"
```
Comment (kattrali) [inline tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/bugsnag_notify_build.rb]: Should be "No config_file ..."

File: tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/bugsnag_notify_build.rb [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+            return
+          end
+        else
+          config_file = params[:config_file]
+        end
+
+        begin
+          config_hash = XmlSimple.xml_in(config_file)
+        rescue ArgumentError => e
+          raise ArgumentError.new "AndroidManifest.xml file not found, please point to your AndroidManifest file"
```
Comment (kattrali) [inline tools/fastlane-plugin/lib/fastlane/plugin/bugsnag/actions/bugsnag_notify_build.rb]: The message here should indicate that the file could not be read. Fastlane also provides `UI.user_error!` for logging fatal errors.

**Pull request merged**